File: web/src/app/models/demande.ts
export class Demande {
  id!: number;
  sujet!: string;
  description!: string;
  date_creation!: Date;
  etat!: Etat;
}

export enum Etat {
  En_cours = 'En_cours',
  Accepte = 'Accepte',
  Rejete = 'Rejete',
}

File: servicedemandes/src/main/java/com/pfa/servicedemandes/model/Demande.java
package com.pfa.servicedemandes.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@AllArgsConstructor
@Data
@NoArgsConstructor
public class Demande {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String sujet;
    private String description;
    private Date dateCreation;
    @Enumerated(EnumType.STRING)
    private Etat etat;
    private String userId;
}
